### PR TITLE
Add documentation of geometry checker

### PR DESCRIPTION
**Pull request by yjacolin**

### CODE REVIEW

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 1]
```diff
@@ -1,10 +1,46 @@
 |updatedisclaimer|
 
+.. index:: Digitizing, Geometry Checker, Topology
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: some more indices: geometry validity, errors

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 2]
```diff
@@ -1,10 +1,46 @@
...
+
 .. _geometry_checker:
 
 Geometry Checker Plugin
 =======================
 
+Geometry Checker is a powerful plugin to check the geometry validity of a layer.
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: a powerful **Core** plugin (just to remind the user)
to check **and fix**, right?

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 3]
```diff
@@ -1,10 +1,46 @@
...
+The :guilabel:`Geometry Checker` dialog show differents grouped settings in the
+first tab (:guilabel:`settings`):
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: **S**ettings

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 4]
```diff
@@ -1,10 +1,46 @@
...
+* :guilabel:`Topology checks`: checks for duplicates, for features within other
+  features, overlaps smaller than a number, for gaps smaller than a number.
+* :guilabel:`Tolerance`: you can define here the tolerance for the check.
+* :guilabel:`Output vector layer` gives the choice to the user how get the
+  result between modifiy the current layer and create a new layer.
+
+After you are happy with the configuration, you can click on the **[Run]** button.
+
+The results appear in the second tab and as an overview layer of the errors in
+the canvas (its name is *checker*. A table list the :guilabel:`geometry check
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: bracket not closed
Do you mean that results are shown either as a table in the second tab of the dialog and as a new layer in the Layers Panel? (sorry, I have never tried this tool)

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 5]
```diff
@@ -1,10 +1,46 @@
...
+result` with one error by row: the first row is an ID, the second the reason of
+the error, then the coordinates of the error, a value (depending of the type of
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: bracket not closed
Not sure but I think it's "depending **on**", the right expression

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 6]
```diff
@@ -1,10 +1,46 @@
...
+the error and finally the resolution column which indicates the resolution of
+the error. At the bottom of this table, you can export the error into a
+shapfile. At the left, you have the number of the errors and the fixed errors.
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: shapefile

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 7]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+* Gaps: gaps between polygons
+
+You can select a row to see the localisation of the error. You can change this
+behaviour by selecting another action between :guilabel:`error` (default),
+:guilabel:`Feature`, :guilabel:`Don't move`, and |checkbox| `Highlight contour
+of selected features`.
+
+Below the zoom action when clicking on the table row, you can :guilabel:`Show
+the selected features in attribute table`, :guilabel:`Fix selected errors using
+default resolution` and :guilabel:`Fix selected errors, promp for resolution
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: prompt

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 8]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+method`. In the latter, you will see a windows to choose the resolution's method
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: window

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 9]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+among other:
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: not sure but maybe rather "among which:"

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 10]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+
+  * Merge with neighboring polygon with longest shared edge,
+  * Merge with neighboring polygon with largest area,
+  * Merge with neighboring polygon identical attribute value,if any, or leave
+    as it
+  * Delete feature
+  * No action
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: this list does not need to be indented I think

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 11]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+
+.. tip:: **Fix multiple errors**
+   You can fix multiple errors by selecting more than one row in the table with
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: Looking at the GitHub preview, wonder if a blank line is not needed before this line

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst [hunk 12]
```diff
@@ -15,3 +51,48 @@ Geometry Checker Plugin
...
+   the *CTRL + click* action.
```
Comment (DelazJ) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: Miss how to fix the errors, doesn't it?

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst [hunk 13]
```diff
@@ -177,6 +177,11 @@ Another option is to use |checkbox| :guilabel:`Enable snapping on intersection`.
 It allows you to snap on an intersection of background layers, even if there's no vertex on
 the intersection.
 
+Geometry Checker
+-----------------
+
+A core plugin can help the user to find the geometry invalidity. You can find
+more information on these plugins at :ref:`geometry_checker`.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: this plugin

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst (hunk 4 above)
Comment (yjacolin) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: exactly!

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst (hunk 5 above)
Comment (yjacolin) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: I have definitively a problem with bracket :) I think you are right on depending **on**

File: source/docs/user_manual/plugins/plugins_geometry_checker.rst (hunk 12 above)
Comment (yjacolin) [inline source/docs/user_manual/plugins/plugins_geometry_checker.rst]: May be I need to rephrase it but the above items list 5 resolution's method. "Resolution" is the term used in the interface, but it is similar to "Fix".

File: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst (hunk 13 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst]: ah yes,  I started to write something for the two core plugins (checker and snapper).

**Pull request merged**